yt_dlp/extractor/nhk.py

```diff
@@ -591,102 +592,198 @@ class NhkRadiruIE(InfoExtractor):
+
+    def _format_actlist(self, act_list):
+        role_groups = {}
+        for act in act_list:
+            role = act.get('role')
+            if role not in role_groups:
+                role_groups[role] = []
+            role_groups[role].append(act)
+
+        formatted_roles = []
+        for role, acts in role_groups.items():
+            for i, act in enumerate(acts):
+                res = f'【{role}】' if i == 0 and role is not None else ''
+                if title := act.get('title'):
+                    res += f'{title}…'
+                res += act.get('name')
+                formatted_roles.append(res)
+        return join_nonempty(*formatted_roles, delim='，')
+
+    def _fetch_artists(self, by_artist):
+        if not by_artist or len(by_artist) == 0:
+            return None
+        artists = []
+        for artist in by_artist:
+            res = ''
+
+            name = artist.get('name')
+            role = artist.get('role')
+            part = artist.get('part')
+
+            if role != '':
+                res += f'{role}…'
+            if part != '':
+                res += f'（{part}）'
+            res += name
+
+            if res != '':
+                artists.append(res)
+        if len(artists) == 0:
+            return None
+        return '、'.join(artists)
+
+    def _fetch_duration(self, duration):
+        d = parse_duration(duration)
+        if d is None:
+            return None
+        hours, remainder = divmod(d, 3600)
+        minutes, seconds = divmod(remainder, 60)
+        res = '（'
+        if hours > 0:
+            res += f'{int(hours)}時間'
+        if minutes > 0:
+            res += f'{int(minutes)}分'
+        res += f'{int(seconds):02}秒）'
+        return res
+
+    def _format_musiclist(self, music_list):
+        if not music_list or len(music_list) == 0:
+            return None
+
+        tracks = []
+
+        for track in music_list:
+            track_details = []
+
+            if name := track.get('name'):
+                track_details.append(f'「{name}」')
+            if lyricist := track.get('lyricist'):
+                track_details.append(f'{lyricist}:作詞')
+            if composer := track.get('composer'):
+                track_details.append(f'{composer}:作曲')
+            if arranger := track.get('arranger'):
+                track_details.append(f'{arranger}:編曲')
+            track_details.append(self._fetch_artists(track.get('byArtist')))
+            track_details.append(self._fetch_duration(track.get('duration')))
+
+            if track.get('label') or track.get('code'):
+                track_details.append('＜' + join_nonempty('label', 'code', delim=' ', from_dict=track) + '＞')
+
+            if location := track.get('location'):
+                track_details.append(f'～{location}～')
+
+            tracks.append(join_nonempty(*track_details, delim='\n'))
+
+        return '\n\n'.join(tracks)
+
```

Comment:
same concept as above with `_get_artists`
```suggestion
    def _format_music_list(self, music_list):
        tracks = []
        for track in traverse_obj(music_list, (..., {dict})):
            track_details = traverse_obj(track, (
                ..., ('name', filter, {'「{}」'.format}),
                ('lyricist', filter, {'{}:作詞'.format}),
                ('composer', filter, {'{}:作曲'.format}),
                ('arranger', filter, {'{}:編曲'.format}),
            ))
            track_details.append(self._fetch_artists(track, 'byArtist'))
            track_details.append(self._fetch_duration(track, 'duration'))
            if label := join_nonempty('label', 'code', delim=' ', from_dict=track):
                track_details.append(f'＜{label}＞')
            if location := traverse_obj(track, ('location', {str})):
                track_details.append(f'～{location}～')

            tracks.append(join_nonempty(*track_details, delim='\n'))

        return '\n\n'.join(tracks)

```